### ISSUE TITLE
DaskExecutor doesn't write results

### ISSUE THREAD

**Issue by ahirner** [PrefectHQ/prefect#3071]
## Description
DaskExecutor doesn't write LocalResults with JSONSerializer.

In our tests, also other results and with a cluster aren't written. For the Repro, I picked the simplest case.
## Expected Behavior
DaskExecutor writes LocalResults. If workers need extra treatment to set their global checkpointing config, I'd consider it an enhancement if configs percolate.

## Reproduction
```python
import json
import os
from tempfile import TemporaryDirectory
from typing import Optional

from prefect import Flow, task
from prefect.engine.executors import Executor, DaskExecutor
from prefect.engine.results import LocalResult
from prefect.engine.serializers import JSONSerializer
from prefect.utilities.configuration import set_temporary_config
from prefect.utilities.debug import raise_on_exception

def test(e: Optional[Executor]):
    with TemporaryDirectory() as tmpdir:
        flow_result = LocalResult(tmpdir, serializer=JSONSerializer(),
                                  location="{task_name}.json")

        with Flow("write_result", result=flow_result) as f:
            _terminal = task(lambda: 42, checkpoint=True, name="magic")()

        with set_temporary_config({"flows.checkpointing": True}), \
             raise_on_exception():
            f.run(executor=e)

        files = os.listdir(tmpdir)
        assert files == ["magic.json"], files
        with open(os.path.join(tmpdir, files[0]), "rb") as file:
            val = json.load(file)
        assert val==42

if __name__ == "__main__":
    print("Local")
    test(None)
    print("DaskExecutor")
    test(DaskExecutor())
```
*Output:*
```bash
Local
[2020-08-01 09:48:05] INFO - prefect.FlowRunner | Beginning Flow run for 'write_result'
[2020-08-01 09:48:05] INFO - prefect.FlowRunner | Starting flow run.
[2020-08-01 09:48:05] INFO - prefect.TaskRunner | Task 'magic': Starting task run...
[2020-08-01 09:48:05] INFO - prefect.TaskRunner | Task 'magic': finished task run for task with final state: 'Success'
[2020-08-01 09:48:05] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
DaskExecutor
[2020-08-01 09:48:05] INFO - prefect.FlowRunner | Beginning Flow run for 'write_result'
[2020-08-01 09:48:05] INFO - prefect.FlowRunner | Starting flow run.
[2020-08-01 09:48:07] INFO - prefect.TaskRunner | Task 'magic': Starting task run...
[2020-08-01 09:48:07] INFO - prefect.TaskRunner | Task 'magic': finished task run for task with final state: 'Success'
[2020-08-01 09:48:08] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
Traceback (most recent call last):
  File "repro_local_write.py", line 37, in <module>
    test(DaskExecutor())
  File "repro_local_write.py", line 26, in test
    assert files == ["magic.json"], files
AssertionError: []
```

## Environment
```json
{
  "config_overrides": {},
  "env_vars": [],
  "system_information": {
    "platform": "Darwin-19.5.0-x86_64-i386-64bit",
    "prefect_version": "0.12.5",
    "python_version": "3.6.8"
  }
}
```
`dask==2.21.0`

**Comment by joshmeek** [comment]
Hi @ahirner I don't think setting temporary config like that will do what you expect due to the workers being created outside of that context manager. Having checkpointing set in your global config.toml (or through an env var) should do the trick:

```
[flows]
checkpointing = true
```

**Comment by ahirner** [comment]
Global configuration options work, thanks! I didn't expect clients to have only partial control over checkpointing.

**Comment by jcrist** [comment]
Hi @ahirner, this was a bug in the task runner code - the flow runner config is shipped to each dask worker during a flow execution, but we were accidentally checking the local config (in the dask worker) for checkpointing rather than the flow runner config. Should be fixed in #3085.

**Comment by ahirner** [comment]
That sounds great @jcrist. 
In our suite, your PR fixes tests with multiprocessing and distributed clients. Thanks a lot. I also think shipping configs is a very user friendly way to control dask clusters, especially given https://github.com/dask/distributed/issues/2908
